android/src/net/hockeyapp/android/NativeCrashManager.java

```diff
@@ -1,100 +1,180 @@
+  public static void handleDumpFiles(Activity activity, String identifier, String installId) {
+    PackageDetails.loadFromContext(activity);
     String[] filenames = searchForDumpFiles();
     for (String dumpFilename : filenames) {
-      String logFilename = createLogFile();
-      if (logFilename != null) {
-        uploadDumpAndLog(activity, identifier, dumpFilename, logFilename);
-      }
+      uploadCrashReport(activity, identifier, dumpFilename, installId);
     }
   }
-  
-  public static String createLogFile() {
-    final Date now = new Date();
 
+  public static String getCurrentDateInAppCenterFormat() {
+    SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+    try{
+      return sdf.format(new Date());
+    }
+    catch(Exception pe){
+      return "";
+    }
+  }
+
+  public static JSONObject getDeviceInfo() {
+
+    JSONObject deviceObj = new JSONObject();
     try {
-      // Create filename from a random uuid
-      String filename = UUID.randomUUID().toString();
-      String path = Constants.FILES_PATH + "/" + filename + ".faketrace";
-      Log.d(Constants.TAG, "Writing unhandled exception to: " + path);
-      
-      // Write the stacktrace to disk
-      BufferedWriter write = new BufferedWriter(new FileWriter(path));
-      write.write("Package: " + Constants.APP_PACKAGE + "\n");
-      write.write("Version Code: " + Constants.APP_VERSION + "\n");
-      write.write("Version Name: " + Constants.APP_VERSION_NAME + "\n");
-      write.write("Android: " + Constants.ANDROID_VERSION + "\n");
-      write.write("Manufacturer: " + Constants.PHONE_MANUFACTURER + "\n");
-      write.write("Model: " + Constants.PHONE_MODEL + "\n");
-      write.write("Date: " + now + "\n");
-      write.write("\n");
-      write.write("MinidumpContainer");
-      write.flush();
-      write.close();
-      
-      return filename + ".faketrace";
-    } 
-    catch (Exception another) {
+      deviceObj.put("appVersion", PackageDetails.APP_VERSION);
+      deviceObj.put("appBuild", "1");
+      deviceObj.put("sdkName", "appcenter.android");
+      deviceObj.put("sdkVersion", "1.0.0");
+      deviceObj.put("osName", "android");
+      deviceObj.put("osVersion", PackageDetails.ANDROID_VERSION);
+      deviceObj.put("model", PackageDetails.PHONE_MODEL);
+
+      deviceObj.put("locale", "en-US");
+    } catch (JSONException e) {
+      e.printStackTrace();
     }
-    
-    return null;
+    return deviceObj;
   }
-  
-  public static void uploadDumpAndLog(final Activity activity, final String identifier, final String dumpFilename, final String logFilename) {
+
+  public static JSONArray createJsonParams(String dumpFileName) {
+    JSONArray jsonArray = new JSONArray();
+    try {
+
+      String currentDateTime = getCurrentDateInAppCenterFormat();
+      String errorId = UUID.randomUUID().toString();
+      String attachmentId = UUID.randomUUID().toString();
+
+      JSONObject logObj = new JSONObject();
```

Comment:
According to https://docs.microsoft.com/en-us/appcenter/diagnostics/upload-crashes the following data are required:

processId: required integer with process identifier
appNamespace: `com.eegeo.recce` or `com.eegeo.mobileexampleapp` depending on the applicationId that we are building

Reply:
These appear to be optional.